state/protocol/badger/validity.go

```diff
@@ -343,6 +348,34 @@ func validateClusterQC(cluster protocol.Cluster) error {
 	return nil
 }
 
+func validateVersionBeacon(snap protocol.Snapshot) error {
```

Comment:
Suggestion: use the `InvalidServiceEventError` as the expected error type for this function. For cases where the input fails validation, return this sentinel rather than a generic error.